.github/workflows/steam.yml

```diff
@@ -0,0 +1,214 @@
+
+    - name: Get build information
+      run: |
+        EVENT='${{ github.event_name }}'
+        if [[ ${EVENT} == 'release' || ( ${EVENT} == 'workflow_dispatch' && -n '${{ github.event.inputs.tag }}') ]]; then
+            if [[ ${EVENT} == "release" ]]; then
+                DESC='${{ github.event.release.tag_name }}'
+                if [ '${{ github.event.release.prerelease }}' == 'true' ]; then
+                  BRANCH='${{ env.STEAM_BETA_BRANCH }}'
+                else
+                  BRANCH='${{ env.STEAM_STABLE_BRANCH }}'
+                fi
+                ASSETS_URL='${{ github.event.release.assets_url }}'
+            else
+                RELEASE=$(curl -s '${{ github.api_url }}/repos/obsproject/obs-studio/releases/tags/${{ github.event.inputs.tag }}')
+                
+                DESC=$(jq -r '.tag_name' <<< "$RELEASE")
+                if [[ $(jq -r '.prerelease' <<< "$RELEASE") == 'true' ]]; then
+                  BRANCH='${{ env.STEAM_BETA_BRANCH }}'
+                else
+                  BRANCH='${{ env.STEAM_STABLE_BRANCH }}'
+                fi
+                ASSETS_URL=$(jq -r '.assets_url' <<< "$RELEASE")
+            fi
+
+            ASSETS=$(curl -s "$ASSETS_URL")
+            WIN_ASSET_URL=$(jq -r '.[] | select(.name|test(".*x64.zip")) .browser_download_url' <<< "$ASSETS")
+            MAC_ASSET_URL=$(jq -r '.[] | select(.name|test(".*.dmg")) .browser_download_url' <<< "$ASSETS")
+            TYPE='release'
+        else
+            BRANCH='${{ env.STEAM_NIGHTLY_BRANCH }}'
+            BUILDS=$(curl -s '${{ github.api_url }}/repos/obsproject/obs-studio/actions/workflows/${{ env.WORKFLOW_ID }}/runs?per_page=1&event=push&status=success&branch=${{ env.GIT_NIGHTLY_BRANCH }}')
+            ARTIFACTS_URL=$(jq -r '.workflow_runs[].artifacts_url' <<< "$BUILDS")
+            DESC="g$(jq -r '.workflow_runs[].head_sha' <<< "$BUILDS" | cut -c1-9)"
+
+            ARTIFACTS=$(curl -s $ARTIFACTS_URL)
+            WIN_ASSET_URL=$(jq -r '.artifacts[] | select(.name|test(".*win-x64.*")) .archive_download_url' <<< "$ARTIFACTS")
+            MAC_ASSET_URL=$(jq -r '.artifacts[] | select(.name|test(".*macos-x86_64.*")) .archive_download_url' <<< "$ARTIFACTS")
+            TYPE='nightly'
+        fi
+
+        # Apply overrides from workflow_dispatch
+        if [[ ${EVENT} == 'workflow_dispatch' ]]; then
+            if [[ -n '${{ github.event.inputs.win_url_override }}' ]]; then
+                WIN_ASSET_URL='${{ github.event.inputs.win_url_override }}'
+            fi
+            
+            if [[ -n '${{ github.event.inputs.mac_url_override }}' ]]; then
+                MAC_ASSET_URL='${{ github.event.inputs.mac_url_override }}'
+            fi
+        fi
+
+        if [[ -z $WIN_ASSET_URL || -z $MAC_ASSET_URL ]]; then
+            echo "Missing at least one asset URL!"
+            exit 1
+        fi
+
+        # set env variables for subsequent steps
+        echo "UPLOAD_TYPE=${TYPE}" >> $GITHUB_ENV
```

Comment:
This is also fine as-is, I just started to prefer setting using step output variables instead of polluting the environment (especially as step outputs can also be re-used as job outputs and then also used in later jobs).

Reply:
I converted this to use outputs instead (although there's only one job, so reusability is kind of a moot point).